### PR TITLE
Fix deprecations with nt.names and kw.data

### PR DESCRIPTION
**Pull request by JeffFessler**
Addresses #200

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -240,9 +240,9 @@ function AxisArray(A::AbstractArray{T,N}, names::NTuple{N,Symbol}, steps::NTuple
 end
 
 # Alternative constructor, takes names as keywords:
-AxisArray(A; kw...) = AxisArray(A, nt_to_axes(kw.data)...)
-@generated nt_to_axes(nt::NamedTuple) =
-    Expr(:tuple, (:(Axis{$(QuoteNode(n))}(getfield(nt, $(QuoteNode(n))))) for n in nt.names)...)
+AxisArray(A; kw...) = AxisArray(A, nt_to_axes(values(kw)))
+@generated nt_to_axes(nt::NamedTuple{names,T}) where {names,T} =
```
Comment (devmotion) [inline src/core.jl:244]: Since `T` is not used you could also write

```suggestion
@generated nt_to_axes(nt::NamedTuple{names}) where {names} =
```

Comment (JeffFessler) [reply src/core.jl]: cool, thanks!

**Comment by rikhuijzer** [comment]
@johnnychen94 could you take a look at this PR? :slightly_smiling_face:  I'd like to run Turing on Julia 1.7.

**Comment by johnnychen94** [comment]
Sorry for the delay! I'll go through other PRs later tonight.

**Pull request merged**